crates/core/src/delta_datafusion/table_provider.rs

```diff
@@ -488,13 +485,12 @@ impl<'a> DeltaScanBuilder<'a> {
             logical_schema
         };
 
-        let context = SessionContext::new();
         let df_schema = logical_schema.clone().to_dfschema()?;
 
         let logical_filter = self
             .filter
             .clone()
-            .map(|expr| simplify_expr(&context, &df_schema, expr));
+            .map(|expr| simplify_expr(self.session, &df_schema, expr));
```

Comment:
This is a drive-by fix and a good example of what @rtyler raised w.r.t. LogStores. We were creating a new session while also tracking a session on the operation. i.e. using inconsistent datafusion sessions in the same operation.